src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/ArtifactsSize.cs

```diff
@@ -0,0 +1,105 @@
+    public void ArtifactsSizeTest()
+    {
+        Assert.True(Directory.Exists(BaselineHelper.GetBaselineFilePath("ArtifactsSizes/")));
+        Assert.NotNull(Config.SourceBuiltArtifactsPath);
+        Assert.NotNull(Config.SdkTarballPath);
+        Assert.NotNull(Config.TargetRid);
+
+        IEnumerable<TarEntry> artifactsTarEntries = Utilities.GetTarballContent(Config.SourceBuiltArtifactsPath).Where(entry => entry.EntryType == TarEntryType.RegularFile);
+        IEnumerable<TarEntry> sdkTarEntries = Utilities.GetTarballContent(Config.SdkTarballPath).Where(entry => entry.EntryType == TarEntryType.RegularFile);
+
+        string[] tarEntries = sdkTarEntries.Concat(artifactsTarEntries)
+            .Select(entry =>
+            {
+                string modifiedPath = buildVersionPattern.Replace(entry.Name, "VERSION");
+                string result = modifiedPath.Replace(Config.TargetRid, "TARGET_RID");
+                return $"{result}: {entry.Length} bytes";
+            })
+            .OrderBy(entry => entry)
+            .ToArray();
+
+        foreach (string entry in tarEntries)
+        {
+            string tarEntryFilename = entry.Substring(0, entry.IndexOf(":")).Trim();
+            string baselineEntry = baselineFileContent?.FirstOrDefault(baselineEntry => baselineEntry.StartsWith(tarEntryFilename)) ?? "";
+
+            if (string.IsNullOrEmpty(baselineEntry))
+            {
+                LogWarningMessage($"{tarEntryFilename} does not exist in baseline. Adding it to the baseline file");
+                File.AppendAllText(baselineFilePath, $"{entry}" + Environment.NewLine);
+            }
+            else
+            {
+                string tarEntrySize = entry.Substring(entry.IndexOf(":") + 1).Replace(" bytes", "").Trim();
+                string baselineEntrySize = baselineEntry.Substring(tarEntryFilename.Length + 1).Replace(" bytes", "").Trim() ?? "0";
+
+                string message = CompareFileSizes(tarEntryFilename, long.Parse(tarEntrySize), long.Parse(baselineEntrySize));
+                if (!string.IsNullOrEmpty(message))
+                {
+                    LogWarningMessage(message);
+                }
+            }
+        }
+
+        CopyBaselineFile();
+    }
+
+    private string CompareFileSizes(string filePath, long fileSize, long baselineSize)
+    {
+        if (fileSize == 0 && baselineSize != 0)
+            return $"{filePath} is now 0 bytes. It was {baselineSize} bytes";
+        else if (fileSize != 0 && baselineSize == 0)
+            return $"{filePath} is no longer 0 bytes. It is now {fileSize} bytes";
+        else if (baselineSize != 0 && Math.Abs(((fileSize - baselineSize) / (double)baselineSize) * 100) >= 25)
+            return $"{filePath} increased in size by more than 25%. It was originally {baselineSize} bytes and is now {fileSize} bytes";
+        return "";
+    }
+
+    private void LogWarningMessage(string message)
```

Comment:
Consider defining this as a common method by moving it to the `Utilities` class. You could even define it as an [extension method](https://learn.microsoft.com/dotnet/csharp/programming-guide/classes-and-structs/extension-methods) of `ITestOutputHelper` which would be pretty nice.

Reply:
Added [here ](https://github.com/dotnet/installer/pull/17233/files#diff-3e52a7713dbdb53496cf7dc21844b2eb5e5934d94c9cd8f54d30106af2f0d21fR112)and implemented [here](https://github.com/dotnet/installer/pull/17233/files#diff-ce0a9d7a18f02288a0510a8ec86e3a5f387ad96613e0296f434de93f02c18da7R94).